src/cli/cli_coap.cpp

```diff
@@ -301,7 +301,7 @@ otError Coap::ProcessRequest(int argc, char *argv[])
         SuccessOrExit(error = otMessageAppend(message, argv[4], payloadLength));
     }
 
-    memset(&messageInfo, 0, sizeof(messageInfo));
+    otMessageInfoInit(&messageInfo);
```

Comment:
There's something nice about being able to initialize a structure simply be mem-zero'ing it. Is there a concrete use case for specifying an IPv6 Hop Limit of 0?

Reply:
From section IPV6_UNICAST_HOPS and IPV6_MULTICAST_HOPS of http://man7.org/linux/man-pages/man7/ipv6.7.html, `0` is a valid value, and `-1` means using the default value. And it seems it's legal to send IP packet to the node itself with hop limit 0. I'm not sure if there's concrete use case for OpenThread. And I'm fine to revert the change that allows hop limit 0.

Reply:
@LuDuda, any thoughts or preference?